### DISCUSSION TITLE
How to open termux from my app?

### DISCUSSION THREAD

**Discussion by nguyenthanhsonbs** [termux/termux-app#2258] [Q&A] [answered]
Hi everyone, thanks for coming here.
I just have a question. I really want to open Termux app from my own app? But Termux will "cd.." or locate to specified path. 

Thanks guy,

**Comment by agnostic-apollo** [comment, marked as answer]
https://github.com/termux/termux-app/wiki/RUN_COMMAND-Intent

**Comment by nguyenthanhsonbs** [comment]
Hi thanks @agnostic-apollo 
I did like that 
1. Request permission:
>ActivityCompat.requestPermissions(Activity.this, new String[]{"com.termux.permission.RUN_COMMAND"}, REQUEST_CODE);
2. Start intent

> Intent intent = new Intent();
> intent.setClassName("com.termux", "com.termux.app.RunCommandService");
> intent.setAction("com.termux.RUN_COMMAND");
> intent.putExtra("com.termux.RUN_COMMAND_PATH", "/data/data/com.termux/files/usr/bin/top");
> intent.putExtra("com.termux.RUN_COMMAND_ARGUMENTS", new String[]{"-n", "5"});
> intent.putExtra("com.termux.RUN_COMMAND_WORKDIR", "/data/data/com.termux/files/home");
> intent.putExtra("com.termux.RUN_COMMAND_BACKGROUND", false);
> intent.putExtra("com.termux.RUN_COMMAND_SESSION_ACTION", "0");
>if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
>   startForegroundService(intent);
>} else {
>   startService(intent);
>}

But nothing happens at all. I miss anything?

**Comment by nguyenthanhsonbs** [comment]
Oh I just found it works only with Termux from FDroid
